datacube/utils/geometry/_base.py

```diff
@@ -213,11 +216,16 @@ def to_epsg(self) -> Optional[int]:
         """
         EPSG Code of the CRS or None
         """
+        if self._epsg is not False:
```

Comment:
Check for `False` instead of None as None is a valid response from `to_epsg`.

Reply:
So False is supposed to mean "don't know epsg yet"?  

In that case _make_crs needs to return epsg=False by default (instead of None).